src/lib.rs

```diff
@@ -83,23 +69,23 @@ pub fn program_from_file(bin_path: &str) -> Vec<U256> {
 }
 
 /// Run a vm program with a clean VM state.
-pub fn run_program(bin_path: &str) -> (U256, VMState) {
-    let vm = VMState::new();
-    run_program_with_custom_state(bin_path, vm)
+pub fn run_program(
+    bin_path: &str,
+    mut vm: VMState,
+    tracers: &mut [Box<&mut dyn Tracer>],
+) -> (U256, VMState) {
+    let program_code = program_from_file(bin_path);
+    vm.load_program(program_code);
+    run(vm, tracers)
 }
 
-/// Run a vm program from the given path using a custom state.
-/// Returns the value stored at storage with key 0 and the final vm state.
-pub fn run_program_with_custom_state(bin_path: &str, mut vm: VMState) -> (U256, VMState) {
-    let program = program_from_file(bin_path);
-    vm.load_program(program);
-    run(vm)
-}
-
-pub fn run(mut vm: VMState) -> (U256, VMState) {
+pub fn run(mut vm: VMState, tracers: &mut [Box<&mut dyn Tracer>]) -> (U256, VMState) {
```

Comment:
Why not make Tracers a part of the VM?
I think it makes sense to do so, plus you don't need to pass around a vec/struct of Tracers around for when you want to run them. Plus we could have some extra logic for when to run them, and having a tracer as vm field, will make it easier to access the VM without having to fight with the borrow checker.

Reply:
Good idea!